VoxelSniperCore/src/main/java/com/github/kevindagame/brush/TreeSnipeBrush.java

```diff
@@ -75,6 +64,23 @@ private void printTreeType(final VoxelMessage vm) {
         vm.custom(printout);
     }
 
+    @Override
+    protected boolean actPerform(SnipeData v) {
+        Undo undo = new Undo();
+        for (var position : positions) {
+            IBlock blockBelow = position.getBlock().getRelative(BlockFace.DOWN);
+            IBlockState currentState = blockBelow.getState();
+            undo.put(currentState);
+            blockBelow.setMaterial(VoxelMaterial.GRASS_BLOCK, false);
+
+            undo = this.getWorld().generateTree(position, this.treeType, undo);
```

Comment:
Why do we re-assign undo?

Reply:
Undo is passed as a variable because old logic. I have touched as little logic as possible in this PR